controllers/datafoundation.go

```diff
@@ -158,6 +159,18 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 			r.Log.Info("Found OCS storage consumers, cannot proceed with uninstallation")
 			return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
 		}
+
+		if err := r.get(&r.storageCluster); err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to get storagecluster: %v", err)
+		}
+		if r.storageCluster.Status.Phase == "Ready" {
+			r.Log.Info("deleting storagecluster")
+			if err := r.delete(&r.storageCluster); err != nil {
+				return ctrl.Result{}, fmt.Errorf("unable to delete storagecluster: %v", err)
+			}
+		} else {
+			r.Log.Info("storagecluster is not in ready state, cannot proceed with uninstallation")
```

Comment:
- you need to requeue the event here

Reply:
I don't think it is required because we are watching storagecluster so if any event occurs on storagecluster CR, there will be a request to the reconciler.

Reply:
- I don't think status update results in a reconcile event 🤔 
- Anyways if this is similar to how we do things in deployer, it's fine with me

Reply:
It should trigger the reconcile event. That's the reason we use generation change predicate where we don't want to reconcile if status/metadata field is changed.